services/search/pkg/search/search.go

```diff
@@ -155,3 +156,31 @@ func convertToWebDAVPermissions(isShared, isMountpoint, isDir bool, p *provider.
 	}
 	return b.String()
 }
+
+func extractScope(path string) (*searchmsg.Reference, error) {
+	ref, err := storagespace.ParseReference(path)
+	if err != nil {
+		return nil, err
+	}
+	return &searchmsg.Reference{
+		ResourceId: &searchmsg.ResourceID{
+			StorageId: ref.ResourceId.StorageId,
+			SpaceId:   ref.ResourceId.SpaceId,
+			OpaqueId:  ref.ResourceId.OpaqueId,
+		},
+		Path: ref.GetPath(),
+	}, nil
+}
+
+// ParseScope extract a scope value from the query string
+func ParseScope(query string) (search, scope string) {
```

Comment:
I don't think there's a general "rule" in the team about that and I know we still have old code using it, but I really think named return variables should be avoided. They allow for confusion and subtle bugs which are hard to track down imo. Do you mind changing this to `func ParseScope(query string) (string, string) {` and explicitly return what you'd like to return?

Reply:
done